crypto/o_str.c

```diff
@@ -223,7 +223,22 @@ int openssl_strerror_r(int errnum, char *buf, size_t buflen)
 #if defined(_MSC_VER) && _MSC_VER>=1400
     return !strerror_s(buf, buflen, errnum);
 #elif defined(_GNU_SOURCE)
-    return strerror_r(errnum, buf, buflen) != NULL;
+    char *err;
+
+    /*
+     * GNU strerror_r may not actually set buf.
+     * It can return a pointer to some (immutable) static string in which case
+     * buf is left unused.
+     */
+    *buf = '\0';
+    err = strerror_r(errnum, buf, buflen);
+    if (err == NULL)
+        return 0;
+    if (*buf == '\0') {
+        strncpy(buf, err, buflen - 1);
```

Comment:
you could use OPENSSL_strlcpy here.

Reply:
I basically just re-used a piece code that was a few lines below this in the #else branch.
I'll update the pull request tomorrow.